lib/postgres/vacuum/monitor/query.rb

```diff
@@ -97,6 +97,28 @@ def blocked_queries
             WHERE NOT blocked_locks.GRANTED;
           SQL
         end
+
+        def connection_state
+          <<-SQL
+            SELECT 
+              state, count(*) as connection_count 
+            FROM pg_stat_activity 
+            GROUP BY state 
+            ORDER BY count DESC;
+          SQL
+        end
+
+        def connection_idle_time
```

Comment:
Did you define this query yourself or find it mentioned somewhere? Might be useful to include a link.

Do we have an idea of how long this query will take to run for a large-ish application like dandy?

Reply:
I wrote it since google didnt help me ='(.

The table itself is tiny. It holds the current activity in the DB which should always be equal to the number of connections open. In dandelion is a bit over 1000.
```
                                                             QUERY PLAN
------------------------------------------------------------------------------------------------------------------------------------
 Limit  (cost=1.29..1.30 rows=1 width=24) (actual time=7.065..7.066 rows=1 loops=1)
   ->  Aggregate  (cost=1.29..1.30 rows=1 width=24) (actual time=7.064..7.064 rows=1 loops=1)
         ->  Function Scan on pg_stat_get_activity s  (cost=0.00..1.25 rows=1 width=8) (actual time=5.913..6.234 rows=1002 loops=1)
               Filter: (state = 'idle'::text)
               Rows Removed by Filter: 41
 Planning Time: 0.313 ms
 Execution Time: 7.205 ms
(7 rows)
```